frontend/src/js/components/subscription/SubscriptionPage.test.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright 2025 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { ADDONS, PLANS } from '@northern.tech/store/appSlice/constants';
+import { act, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { vi } from 'vitest';
+
+import { defaultState, undefineds } from '../../../../tests/mockData';
+import { render } from '../../../../tests/setupTests';
+import { SubscriptionPage } from './SubscriptionPage';
+
+const enterpriseReq = {
+  content: {
+    current_addons: '-',
+    current_plan: 'Basic',
+    requested_addons: 'monitor',
+    requested_plan: 'Enterprise',
+    user_message: 'I want mender so bad'
+  },
+  tenantId: '1'
+};
+
+const professionalReq = {
+  plan: 'professional',
+  planName: 'professional',
+
+  preview_mode: 'recurring',
+  products: [
+    {
+      addons: [],
+      name: 'mender_standard',
+      quantity: 250
+    }
+  ]
+};
+const professionalReq300 = { ...professionalReq, products: [{ ...professionalReq.products[0], quantity: 300 }] };
+const professionalReq300WithMonitor = {
+  ...professionalReq,
+  products: [{ ...professionalReq.products[0], quantity: 300, addons: [{ name: ADDONS.monitor.id }] }]
+};
+describe('Subscription Summary component', () => {
+  it('renders correctly', async () => {
+    const { baseElement } = render(<SubscriptionPage />);
+    const view = baseElement.lastElementChild;
+    expect(view).toMatchSnapshot();
+    expect(view).toEqual(expect.not.stringMatching(undefineds));
+  });
+
+  it('allows signing up', async () => {
+    const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime });
+    const organizationActions = await import('@northern.tech/store/organizationSlice/thunks');
+    const getBillingPreview = vi.spyOn(organizationActions, 'getBillingPreview');
+    const requestEnterprise = vi.spyOn(organizationActions, 'requestPlanChange');
+    const ui = <SubscriptionPage />;
+    render(ui, {
+      preloadedState: {
+        ...defaultState,
+        devices: { ...defaultState.devices, limit: 10 },
+        organization: { ...defaultState.organization, organization: { ...defaultState.organization.organization, trial: true } }
+      }
+    });
+    expect(screen.getByText('Upgrade your subscription')).toBeInTheDocument();
+    const professionalRadioButton = screen.getByRole('radio', { name: new RegExp(PLANS.professional.id, 'i') });
```

Comment:
The accessible `name` here is derived from the `label` of the buttons afaik, which OTOH is set with the `name`:
```suggestion
    const professionalRadioButton = screen.getByRole('radio', { name: new RegExp(PLANS.professional.name, 'i') });
```
works for anything, but 'os'/ 'basic' - but might stop working once the tinier compute units get introduced...